src/js/components/devices/device-details/deployments.js

```diff
@@ -49,34 +53,58 @@ const EmptyState = ({ isFiltered }) => (
   </div>
 );
 
-const columns = [
-  { content: 'Release', key: 'release', Component: ({ deviceDeployment: { release } }) => <Link to={`/releases/${release}`}>{release}</Link> },
-  {
-    content: 'Target device(s)',
-    key: 'target',
-    Component: ({ deviceDeployment: { id, target, route } }) => <Link to={`/deployments/${route}?id=${id}&open=true`}>{target}</Link>
-  },
-  { content: 'Started', key: 'created', Component: ({ deviceDeployment: { created } }) => <MaybeTime value={created} /> },
-  { content: 'Finished', key: 'finished', Component: ({ deviceDeployment: { finished } }) => <MaybeTime value={finished} /> },
-  {
-    content: 'Deployment progress',
-    key: 'deploymentStatus',
-    Component: ({ deviceDeployment: { deploymentStatus } }) => deploymentDisplayStates[deploymentStatus] ?? deploymentStatus
-  },
-  { content: 'Device status', key: 'status', Component: ({ deviceDeployment: { status } }) => status },
-  {
-    content: '',
-    key: 'log',
-    Component: ({ deviceDeployment: { id, log, deviceId } }) =>
-      log && (
-        <Button component="a" href={`${window.location.origin}${deploymentsApiUrl}/deployments/${id}/devices/${deviceId}/log`} target="_blank">
-          Log
-        </Button>
-      )
-  }
-];
-
 const History = ({ className, items, page, perPage, setPage, setPerPage, total }) => {
+  const { token } = useSelector(getCurrentSession);
+  const columns = [
```

Comment:
no need to move columns inside the history component, you can pass token to the component

```
{columns.map(({ key, Component }) => (
                <TableCell key={`${item.id}-${key}`}>
                  <Component deviceDeployment={item} token={token} />
                </TableCell>
              ))}
```

and get it later 
```Component: ({ deviceDeployment: { id, log, deviceId }, token }) =>```